README.md

```diff
@@ -0,0 +1,180 @@
+ssl_cert = ~/sphinx-test/ssl_cert.pem
+EOF
+
+# Run the server
+$ oracle # do not kill this process
+```
+
+## CLI client setup
+
+Setup the client in a new terminal.
+
+```bash
+# Install & activate all dependencies.
+$ nix develop
+# Go to the test folder.
+$ cd ~/sphinx-test
+
+# Setup the client configuration.
+# Note: For a more realistic scenarion, replache 127.0.0.1 by the actual IP
+# address of the localhost. Also note that the same cfg file (see above) is used.
+$ cat <<EOF >> sphinx.cfg # append to the cfg file
+[client]
+verbose = True
+address = 127.0.0.1
+port = 2355
+datadir = ~/sphinx-test/datadir
+ssl_key = ~/sphinx-test/ssl_key.pem
+ssl_cert = ~/sphinx-test/ssl_cert.pem
+EOF
+
+# Generate the master key that is used to derive secrets. This key must be
+# shared among clients.
+$ sphinx init
+$ ls ./datadir/masterkey
+
+# Store credentials. You should see log output from the oracle server and the
+# client should return a password.
+# See https://github.com/stef/pwdsphinx#create-password for details.
+$ printf 'm' | sphinx create user site uld 10 # "m" is the master password
+<password>
+
+# Retrieve credentials.
+printf 'm' | sphinx get user site
+<password>
+printf 'wrongmasterpassword' | sphinx get user site
+<some-other-password>
+```
+
+## Android client setup
+
+Connect your phone to the development machine and make sure the 'Developer options' are enabled.
+
+Open a 3rd terminal.
+
+```bash
+# Install & activate all dependencies.
+$ nix develop
+# Go to the test folder.
+$ cd ~/sphinx-test
+
+# Check that your phone is accessible.
+$ adb devices
+
+# Install the custom SSL certificate.
+adb push ssl_cert.pem mnt/sdcard/ssl_cert.pem # or wherever else you can upload to the phone
+
+# In your phone, open Settings -> Security -> "Install [certificates] from
+# device memory/SD card" (or similar) -> Choose the certificate file and install
+# it.
+
+# Install the app. (The 'uninstall' command will fail at the first time, of
+# course).
+$ ls $DEBUG_APK # see flake.nix
+$ adb uninstall org.hsbp.androsphinx ; adb install $DEBUG_APK
+
+# At this point, you can unplug the phone. Make sure it is connected to the
+# local network instead. Also, get the dev machine's IP address.
+$ ip a
+$ export IP_ADDR=X.X.X.X # 192.168....
+
+# Launch the androsphinx app in the phone, press the 'Settings' icon and press 'Scan from
+# QR code'. Generate the QR code on the dev machine (make sure the $IP_ADDR is set).
+(printf '\x01' ; cat ./datadir/masterkey ; printf '\x09\x33%s' "$IP_ADDR") | qrencode -8 -t ANSI256
+# (Now scan the generated code with the phone.)
+```
+
+The phone is now configured correctly.
+Make sure the server process (``oracle``) is running.
+Use the app's search form to search for "site".
+You should see a log statement from the sphinx server as well as the entry "user" on the phone.
+Try copying the password to the clipboard by using the master password from above.
+You should receive the same password as the CLI client.
+
```

Comment:
You might want to add some update instructions, what repo to pull androsphinx and generate the gradle env from or provide an update script for future contributors.